src/serialize.js

```diff
@@ -133,11 +133,43 @@ function replaceRecord(rec, replaceChild) {
   return { "__record": rec._name, "data": recordData }
 }
 
+function getIterableType(iterable) {
+  if (iterable.constructor === immutable.List) {
+    return 'List'
+  }
+
+  if (iterable.constructor === immutable.Set) {
+    return 'Set'
+  }
+
+  if (iterable.constructor === immutable.OrderedSet) {
+    return 'OrderedSet'
+  }
+
+  if (iterable.constructor === immutable.Stack) {
+    return 'Stack'
+  }
+
+  if (iterable.constructor === immutable.Map) {
+    return 'Map'
+  }
+
+  if (iterable.constructor === immutable.OrderedMap) {
```

Comment:
Nechceš to spíš udělat jako `switch (iterable.constructor)`, aby to bylo kompaktnější?